File: android/app/src/main/java/com/mattermost/rnbeta/NotificationDismissService.java
package com.mattermost.rnbeta;

import android.content.Context;
import android.content.Intent;
import android.app.IntentService;
import android.os.Bundle;

import com.mattermost.rnutils.helpers.NotificationHelper;
import com.mattermost.turbolog.TurboLog;
import com.wix.reactnativenotifications.core.NotificationIntentAdapter;

public class NotificationDismissService extends IntentService {
    public NotificationDismissService() {
            super("notificationDismissService");
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        final Context context = getApplicationContext();
        final Bundle bundle = NotificationIntentAdapter.extractPendingNotificationDataFromIntent(intent);

        NotificationHelper.INSTANCE.dismissNotification(context, bundle);
        TurboLog.Companion.i("ReactNative", "Dismiss notification");
    }
}

File: android/app/src/main/java/com/mattermost/rnbeta/NotificationDismissService.kt
package com.mattermost.rnbeta

import android.app.IntentService
import android.content.Context
import android.content.Intent
import android.util.Log
import com.wix.reactnativenotifications.core.NotificationIntentAdapter

class NotificationDismissService : IntentService("notificationDismissService") {
    private var mContext: Context? = null
    override fun onHandleIntent(intent: Intent) {
        mContext = applicationContext
        val bundle = NotificationIntentAdapter.extractPendingNotificationDataFromIntent(intent)
        val notificationId = intent.getIntExtra(CustomPushNotification.NOTIFICATION_ID, -1)
        val channelId = bundle.getString("channel_id")
        CustomPushNotification.clearNotification(mContext, notificationId, channelId)
        Log.i("ReactNative", "Dismiss notification")
    }
}
